docs/topics/dokka-migration.md

```diff
@@ -0,0 +1,409 @@
+
+In the new version, the `dokkaGenerate` task name works for both single and multi-module projects. You can use different tasks
+to generate output in HTML, Javadoc or both HTML and Javadoc. For more information, see the next section.
+
+### Select documentation output format
+
+> The Javadoc output format is in [Alpha](https://kotlinlang.org/docs/components-stability.html#stability-levels-explained).
+> You may find bugs and experience migration issues when using it. Successful integration with tools that accept Javadoc
+> as input is not guaranteed. Use it at your own risk.
+>
+{style="note"}
+
+With DGP v2, you can choose to generate the API documentation in HTML, Javadoc, or both formats at the same time:
+
+1. Place the corresponding plugin `id` in the `plugins{}` block of your project's `build.gradle.kts` file:
+
+   ```kotlin
+   plugins {
+       // Generates HTML documentation
+       id("org.jetbrains.dokka") version "2.0.0"
+       // OR
+       // Generates Javadoc documentation
+       id("org.jetbrains.dokka-javadoc") version "2.0.0"
+       // Keeping both plugin IDs generates both formats
+   }
+   ```
+
+2. Run the corresponding Gradle task.
+
+Here's a list of the plugin `id` and Gradle task that correspond to each format:
+
+|             | **HTML**                       | **Javadoc**                         | **Both**                          |
+|-------------|--------------------------------|-------------------------------------|-----------------------------------|
+| Plugin `id` | `id("org.jetbrains.dokka")`    | `id("org.jetbrains.dokka-javadoc")` | Use both HTML and Javadoc plugins |
+| Gradle task | `./gradlew :dokkaGenerateHtml` | `./gradlew :dokkaGenerateJavadoc`   | `./gradlew :dokkaGenerate`        |
+
+> The `dokkaGenerate` task generates documentation in all available formats based on the applied plugins. 
+> If both the HTML and Javadoc plugins are applied, you can choose to generate only HTML format by running the `dokkaGenerateHtml` task, 
+> or only Javadoc by running the `dokkaGenerateJavadoc` task.
+> 
+{style="tip"}
+
+### Address deprecations and removals
+
+* **Output format support:** Dokka now only supports HTML and Javadoc output. Experimental formats like Markdown and Jekyll are no longer supported.
+* **Collector task:** `DokkaCollectorTask` has been removed. Now, you need to separate the documentation generation for
+  each subproject and then [aggregate the documentation](#update-documentation-aggregation-in-multi-module-projects) if necessary.
+
+## Finish up your migration
+
+After you've migrated your project, perform these steps to wrap up and improve performance.
+
+### Set the opt-in flag
+
+After successful migration, set the following opt-in flag without helpers in the project's `gradle.properties` file:
+
+```kotlin
+org.jetbrains.dokka.experimental.gradle.pluginMode=V2Enabled
```

Comment:
I think it's because Dokka v2 introduces new helpers to simplify configuration, but I am comfirming to add more context about enabling and then removing.

Reply:
Some brief context, if it helps:

DGPv1 has several tasks, like `tasks.dokkaHtmlMultiModule`. These tasks don't exist in DGPv2.

When users start migrating to DGPv2, if their buildscripts reference these tasks then the buildscripts will fail to compile. These compilation failures are overwhelming and unhelpful.

To avoid compilation errors, users can start gradually migrating to V2 by setting the plugin mode to `V2EnabledWithHelpers`. This mode enables DGPv2, but will also register the same tasks as DGPv2. We call these tasks 'migration helpers'.

The migration helper tasks are non-functional, and cannot execute. They are only there to prevent buildscript errors. DGPv2 will log a nagging warning encouraging users to disable them (by setting the plugin mode to `V2Enabled`.

The migration helpers should be disabled as soon as migration is complete.

Reply:
I see the addition above, I think the sequence is clearer now. Still, I would add a comment after this code block that explicitly says that if you removed references to old Gradle tasks, now you shouldn't see compilation errors related to that. As far as I understand, the problem Adam is talking about is still relevant If someone misses a step, or misunderstands the intro in the beginning.